### PR TITLE
Add paper bloom filter

### PR DESCRIPTION
**Pull request by neithernut**
This PR adds a section about the usage of bloom filters.

### CODE REVIEW

File: paper/bloom-filter.tex [hunk 1]
```diff
@@ -0,0 +1,108 @@
+\section{Bloom filter usage}
+\label{sec:bloom_filter}
+
+    %TODO: reference
+    A bloom filter is a probabilistic data structure which allows a fast check
+    for a potential membership of an item in a set $B$.
+    As a data structure, it consists of an array of a constant number $m$ of
+    bits $b_0$ to $b_{m-1}$.
+    For an empty set, all those bits are set to NULL.
```
Comment (matthiasbeyer) [inline paper/bloom-filter.tex]: Not zero?

File: paper/bloom-filter.tex [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    On insertion of an element into the set, $k$ hash functions $h_0$ to
+    $h_{k-1}$ are applied to the function.
```
Comment (matthiasbeyer) [inline paper/bloom-filter.tex]: Applied to which function? Hash functions are applied on a function?

File: paper/bloom-filter.tex [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    If all the bits relevant for the element are set, the element \emph{may} be
+    in the set $B$:
+
+    \begin{equation}
+        \forall_i : b_{h_i(e)\%m} \Rightarrow
+        p(e \in B) = \left(1 - \left(1 - \frac{1}{m} \right)^{kn} \right)^k
+    \end{equation}
+
+    In Libreset, bloom filters will be used as filters to speed up some
```
Comment (matthiasbeyer) [inline paper/bloom-filter.tex]: "libreset" should be lower case here.

File: paper/bloom-filter.tex [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    The use of bloom filters will, however, exceed the classical use for
+    speeding up lookups.
+    The filter may also be used to speed up merges and intersections by allowing
+    early recognition of disjunctive sets.
+
+    \subsection{Disjunctive sets}
+    \label{sec:bloom_filter-disjunctive_sets}
+
+        Checking whether the intersection of two sets $A$ and $B$ is not empty
+        is equivalent to checking whether at least one element of set $A$ is in
```
Comment (matthiasbeyer) [inline paper/bloom-filter.tex:44]: The "least" one? You meant the "last" one, don't you?

File: paper/bloom-filter.tex (hunk 1 above)
Comment (neithernut) [inline paper/bloom-filter.tex]: Good point.

File: paper/bloom-filter.tex (hunk 2 above)
Comment (neithernut) [inline paper/bloom-filter.tex]: Mh... I edited that sencence once too often...

File: paper/bloom-filter.tex (hunk 3 above)
Comment (neithernut) [inline paper/bloom-filter.tex]: It's a name.

File: paper/bloom-filter.tex (hunk 4 above)
Comment (neithernut) [inline paper/bloom-filter.tex:44]: _at least_ one. German: _mindestens_ eins

File: paper/bloom-filter.tex (hunk 3 above)
Comment (matthiasbeyer) [inline paper/bloom-filter.tex]: And the name is lowercase in all letters.

File: paper/bloom-filter.tex (hunk 4 above)
Comment (matthiasbeyer) [inline paper/bloom-filter.tex:44]: My fault here...

File: paper/bloom-filter.tex (hunk 3 above)
Comment (neithernut) [inline paper/bloom-filter.tex]: That's odd, but I'll "fix" this.

**Comment by neithernut** [comment]
Do I have a GO for squashing?

**Comment by matthiasbeyer** [comment]
yep.

**Comment by neithernut** [comment]
Done. May now be merged.

**Pull request merged**